### PR TITLE
(Closes #2538) Added a preprocessor interface

### LINKED ISSUE #2538: Create a PreprocessorInterface
To be able to use things like __FILE__ and __LINE__ for otter we need to create variables that won't appear in a symbol table or a wildcard import.
Sergi's suggestion is to create a PreprocessorInterface that ignores some of the other rules.

Starting point:
https://github.com/stfc/PSyclone/compare/master...1588_jit_opencl_invariants

### PR DESCRIPTION
**Pull request by LonelyCat124**

### CODE REVIEW

**Comment by LonelyCat124** [comment]
@sergisiso this is done, it doesn't really add a lot by itself but it should be useful for the Otter stuff.

File: src/psyclone/psyir/symbols/interfaces.py [hunk 1]
```diff
@@ -252,3 +252,10 @@ def copy(self):
         :rtype: :py:class:`psyclone.psyir.symbol.SymbolInterface`
         '''
         return self.__class__(access=self.access)
+
+
+class PreprocessorInterface(SymbolInterface):
```
[the remark this answers is not in the document]
Comment (LonelyCat124) [reply src/psyclone/psyir/symbols/interfaces.py:257]: Added

File: src/psyclone/tests/psyir/symbols/interfaces_test.py [hunk 2]
```diff
@@ -237,3 +237,9 @@ def test_argumentinterface_copy():
     # Check that we can modify the copy without affecting the original
     new_interface.access = ArgumentInterface.Access.READ
     assert arg_interface.access == ArgumentInterface.Access.WRITE
+
+
+def test_preprocessorinterface():
+    ''' Test the PreprocessorInterface method.'''
+    interface = PreprocessorInterface()
+    assert str(interface) == "Preprocessor"
```
[the remark this answers is not in the document]
Comment (LonelyCat124) [reply src/psyclone/tests/psyir/symbols/interfaces_test.py:245]: Added a check to this test.

**Comment by LonelyCat124** [comment]
@sergisiso Should have added the changes you requested here now.

**Comment by LonelyCat124** [comment]
Hmm, the test isn't currently working so I'll check this tomorrow.

**Comment by LonelyCat124** [comment]
@sergisiso Ok, I had to add something to the backend to ignore PreprocessorInterface symbols but this should be ready for another look now.

**Review by sergisiso — APPROVED** [review]
All looks good now @LonelyCat124 . Tests and documentation have been added, there is no linting issues and the coverage is complete. This is ready to merge.

**Pull request merged**